redaxo/src/core/lib/debug_bar/debug_bar_collector.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+/**
+ * Class for debug bar
+ *
+ * @author gharlan
+ *
+ * @package redaxo\core
+ */
+abstract class rex_debug_bar_collector
+{
+    /**
+     * Returns the html bar item
+     *
+     * @return string
+     */
+    abstract public function getBarItem();
+
+    /**
+     * Returns the position in the debug bar
+     *
+     * @return bool
+     */
+    public function onLeftSide()
+    {
+        return true;
+    }
+
+    /**
+     * Returns the danger status
+     *
+     * @return bool
+     */
+    public function isDanger()
+    {
+        return false;
+    }
+
+    /**
+     * Returns the primary status
```

Comment:
Was bedeutet primary in diesem kontext?

Reply:
Das hat nur farbliche Relevanzen.  Genauso wie danger und warning.

Wenn primary true ist, bekommt das Item in der Leiste einen hellblauen Hintergrund.

https://github.com/tbaddade/redaxo/blob/tbaddade-debug-toolbar/redaxo/src/addons/be_style/plugins/redaxo/scss/_debug_bar.scss#L123-L125

Reply:
siehe auch hier im [Screen](https://github.com/redaxo/redaxo/issues/1909) die Versionsangabe rechts.